[JSC] YARR: Update UCS canonicalization tables

tetsuharuohzeki opened the pull request:
#### 99be77af7cd26a3cccfd2d037e0f0d2478d43513
<pre>
[JSC] YARR: Update UCS canonicalization tables
<a href="https://bugs.webkit.org/show_bug.cgi?id=286687">https://bugs.webkit.org/show_bug.cgi?id=286687</a>

Reviewed by Michael Saboff.

test262 has coverage for this change.

* Source/JavaScriptCore/yarr/YarrCanonicalize.h:
(JSC::Yarr::canonicalRangeInfoFor):
* Source/JavaScriptCore/yarr/YarrCanonicalizeUCS2.cpp: I made this change by:
    `/System/Library/Frameworks/JavaScriptCore.framework/Versions/Current/Helpers/jsc ./YarrCanonicalizeUCS2.js &gt; ./YarrCanonicalizeUCS2.cpp`
  on macOS 15.3（24D60).
* Source/JavaScriptCore/yarr/YarrCanonicalizeUCS2.js: Rename some generated variable names to match our style checker.
(printHeader):
(printFooter):
(set characters.hex.set string_appeared_here):
(set characters):
(characters.string_appeared_here.set characterSetInfo):

Canonical link: <a href="https://commits.webkit.org/290334@main">https://commits.webkit.org/290334@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/057d89597e6a9ec0e7eaf9efa8a0f21e140a4f5c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/89520 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/9047 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/44377 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/94512 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/40287 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/9436 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/17326 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/68956 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/26612 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/92521 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/7241 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/81242 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/49322 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/6972 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/35641 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/39393 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/82318 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/77326 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/36627 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/96340 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/88295 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/16702 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/12355 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/77922 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/16957 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/77043 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/77144 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19076 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/21572 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/20151 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/9862 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/16715 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/22028 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/110788 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/16456 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/26581 "Found 5 new JSC stress test failures: wasm.yaml/wasm/function-tests/memcpy-wasm-loop.js.wasm-slow-memory, wasm.yaml/wasm/stress/repro_1289.js.default-wasm, wasm.yaml/wasm/stress/repro_1289.js.wasm-eager, wasm.yaml/wasm/stress/top-most-enclosing-stack.js.wasm-collect-continuously, wasm.yaml/wasm/stress/top-most-enclosing-stack.js.wasm-eager-jettison (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/19907 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/18238 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

tetsuharuohzeki commented on Source/JavaScriptCore/yarr/YarrCanonicalizeUCS2.cpp:
```diff
@@ -81,7 +81,7 @@ const char32_t* const ucs2CharacterSetInfo[UCS2_CANONICALIZATION_SETS] = {
 };
 
-const size_t UCS2_CANONICALIZATION_RANGES = 448;
+const size_t UCS2_CANONICALIZATION_RANGES = 460;
```
style-checker warn the following error here but this source is generated code.

```
ERROR: Source/JavaScriptCore/yarr/YarrCanonicalizeUCS2.cpp:84:  UCS2_CANONICALIZATION_RANGES is incorrectly named. Don't use underscores in your identifier names.  [readability/naming/underscores] [4]
```

darinadler replied:
Typically when we run into something like this, we update the style checker script so it knows about exceptions, or tweak the script to change the generated code.

tetsuharuohzeki replied:
I see. I updated the codegen to match the style checker.

darinadler reviewed (commented):
Do we have tests for what is changing? Generally we don’t want to just change the code, even if we are certain the change is correct.

msaboff reviewed (approved):
r=me
Thanks for taking care of this.

You should add to the change log that test262 has coverage for this change.

darinadler commented on Source/JavaScriptCore/yarr/YarrCanonicalizeUCS2.js:
```diff
@@ -58,13 +60,13 @@ function printHeader()
     print('#include "YarrCanonicalize.h"');
     print();
-    print("namespace JSC { namespace Yarr {");
+    print("namespace JSC::Yarr {");
```
Why this change? Doesn’t seem helpful.

tetsuharuohzeki replied:
I aimed to sort the style with files that added recently and I did not found any restriction about nested namespace short syntax introduced in C++17 that in [our coding guideline](https://webkit.org/code-style-guidelines/#indentation-namespace). If it's a problem, I'll fix it. How about that?

- https://github.com/WebKit/WebKit/blob/80f5cfefbfe9c625416a0007849ab8fa9c6294b5/Source/WebCore/Modules/WebGPU/Implementation/WebGPUDowncastConvertToBackingContext.cpp#L61
- https://github.com/WebKit/WebKit/blob/80f5cfefbfe9c625416a0007849ab8fa9c6294b5/Source/JavaScriptCore/b3/air/AirHelpers.h#L32

darinadler replied:
Normally we would only change if it's an improvement. Otherwise we are just churning code with competing coding styles. I don't see any connection to the rest of what is done here nor an argument that it's an improvement so I would just leave it alone.

If you think it's an improvement I'm not sure why we would start here.

darinadler replied:
Another way to put it is "there's no rule that says we _can't_ use this syntax" is not enough of an argument in favor of changing existing code. Maybe you do have a reason, but it's usually not a good idea to make a change without citing a reason.

darinadler replied:
Or are you saying that new source files in the project are recently using this style? Anyway, I wouldn't do the change but I won't stand in your way. It's not important either way.

tetsuharuohzeki replied:
I got it. I'll revert this change.

darinadler commented on Source/JavaScriptCore/yarr/YarrCanonicalizeUCS2.js:
```diff
@@ -160,22 +161,25 @@ function createTables(prefix, maxValue, canonicalGroups)
-    print("const size_t " + prefixUpper + "_CANONICALIZATION_RANGES = " + rangeInfo.length + ";");
-    print("const CanonicalizationRange " + prefixLower + "RangeInfo[" + prefixUpper + "_CANONICALIZATION_RANGES] = {");
+    const canonicalizationRangesVarName = `${prefixLower}CanonicalizationRanges`;
+    print(`const size_t ${canonicalizationRangesVarName} = ${rangeInfo.length};`);
+    print(`const CanonicalizationRange ${prefixLower}RangeInfo[${canonicalizationRangesVarName}] = {`);
```
We could use `constexpr` here.

tetsuharuohzeki replied:
I'm planning to do that in https://bugs.webkit.org/show_bug.cgi?id=286686 after we landing this.

tetsuharuohzeki committed:
[JSC] YARR: Update UCS canonicalization tables
https://bugs.webkit.org/show_bug.cgi?id=286687

Reviewed by Michael Saboff.

test262 has coverage for this change.

* Source/JavaScriptCore/yarr/YarrCanonicalize.h:
(JSC::Yarr::canonicalRangeInfoFor):
* Source/JavaScriptCore/yarr/YarrCanonicalizeUCS2.cpp: I made this change by:
    `/System/Library/Frameworks/JavaScriptCore.framework/Versions/Current/Helpers/jsc ./YarrCanonicalizeUCS2.js > ./YarrCanonicalizeUCS2.cpp`
  on macOS 15.3（24D60).
* Source/JavaScriptCore/yarr/YarrCanonicalizeUCS2.js: Rename some generated variable names to match our style checker.
(printHeader):
(printFooter):
(set characters.hex.set string_appeared_here):
(set characters):
(characters.string_appeared_here.set characterSetInfo):

Canonical link: https://commits.webkit.org/290334@main
